meson.build

```diff
@@ -0,0 +1,23 @@
+# Setup the Meson Project
+project('llnl-hatchet', 'cython',
+    version: '2024.1.1'
+)
+
+# Get the Meson Python object, a dependency object to Python for extension modules,
+# and the path to the top-level install directory
+py = import('python').find_installation(pure: false)
+py_dep = py.dependency()
```

Comment:
Two questions related to this @michaelmckinsey1:
1. Do you have `pkgconfig` installed?
2. Where are your Python development libraries (i.e., `libpython<version_stuff>.so`) installed?

Long story short, this is essentially caused by Windows being different from every other OS on the planet and doing certain things differently from every other OS.

Reply:
@michaelmckinsey1 when you get the chance, can you try to download and install the following sdist in a Python venv?

https://files.pythonhosted.org/packages/65/6e/09db70a523a96d25e115e71cc56a6f9031e7b8cd166c1ac8438307c14058/numpy-1.26.4.tar.gz

Once the file is downloaded, you can install with

```
python3 -m pip install numpy-1.26.4.tar.gz
```

Reply:
I'm curious if that will fail with the same error. From what I can tell, the Python lookup Meson code is the same as what NumPy uses. If installing that sdist fails too, then there's likely an issue with your Windows install of Python. But, if it doesn't, then there's something wrong with my Meson stuff in Hatchet.

Reply:
1. I did not have `pkgconfig`, but installing it did not fix the issue.
2. They would be `.dll` not `.so` right? I don't even see any `libpython*.dll`, all I see in conda is:
```
miniconda3/envs/test-meson/Lib/site-packages/Cython/Debugger/libpython.py
miniconda3/envs/test-meson/Lib/site-packages/Cython/Debugger/Tests/test_libpython_in_gdb.py
miniconda3/envs/test-meson/Lib/site-packages/Cython/Debugger/Tests/__pycache__/test_libpython_in_gdb.cpython-311.pyc
miniconda3/envs/test-meson/Lib/site-packages/Cython/Debugger/__pycache__/libpython.cpython-311.pyc
```
3. I tried downloading the sdist and still got the same error

I also tried with `venv`, and I tried on another windows device with conda, same error. I would assume for now it's a general issue with our meson build for Windows, unless someone else is able to successfully build on Windows.

Reply:
If it also happened with the sdist from above, then it's an issue with either conda or Windows, not necessarily our code. The files it is looking for are **not** from Cython. It's looking for the Python development libraries (e.g., `libpython3.9.so` on Linux, `libpython3.9.dylib` on Mac). These should be in your Anaconda install. On UNIX platforms, they're located in `<anaconda root>/lib`. Not sure where they are on Windows.

By default, Meson uses pkgconfig to find libraries. Even for Python, I believe that pkgconfig is its default search mechanism.

It's possible that the issue could be caused by the venv, but I doubt that's the cause.

Reply:
@michaelmckinsey1 can you run `python3 -m sysconfig` and paste the output here?

Reply:
For what it's worth, I can now reproduce on macOS when pkgconfig is not installed

Reply:
```
$ python -m sysconfig
Platform: "win-amd64"
Python version: "3.11"
Current installation scheme: "nt"

Paths:
        data = "C:\Users\Micha\miniconda3\envs\test-meson"
        include = "C:\Users\Micha\miniconda3\envs\test-meson\Include"
        platinclude = "C:\Users\Micha\miniconda3\envs\test-meson\Include"
        platlib = "C:\Users\Micha\miniconda3\envs\test-meson\Lib\site-packages"
        platstdlib = "C:\Users\Micha\miniconda3\envs\test-meson\Lib"
        purelib = "C:\Users\Micha\miniconda3\envs\test-meson\Lib\site-packages"
        scripts = "C:\Users\Micha\miniconda3\envs\test-meson\Scripts"
        stdlib = "C:\Users\Micha\miniconda3\envs\test-meson\Lib"

Variables:
        BINDIR = "C:\Users\Micha\miniconda3\envs\test-meson"
        BINLIBDEST = "C:\Users\Micha\miniconda3\envs\test-meson\Lib"
        EXE = ".exe"
        EXT_SUFFIX = ".cp311-win_amd64.pyd"
        INCLUDEPY = "C:\Users\Micha\miniconda3\envs\test-meson\Include"
        LIBDEST = "C:\Users\Micha\miniconda3\envs\test-meson\Lib"
        TZPATH = "C:\Users\Micha\miniconda3\envs\test-meson\share\zoneinfo"
        VERSION = "311"
        VPATH = "..\.."
        abiflags = ""
        base = "C:\Users\Micha\miniconda3\envs\test-meson"
        exec_prefix = "C:\Users\Micha\miniconda3\envs\test-meson"
        installed_base = "C:\Users\Micha\miniconda3\envs\test-meson"
        installed_platbase = "C:\Users\Micha\miniconda3\envs\test-meson"
        platbase = "C:\Users\Micha\miniconda3\envs\test-meson"
        platlibdir = "DLLs"
        prefix = "C:\Users\Micha\miniconda3\envs\test-meson"
        projectbase = "C:\Users\Micha\miniconda3\envs\test-meson"
        py_version = "3.11.9"
        py_version_nodot = "311"
        py_version_nodot_plat = "311"
        py_version_short = "3.11"
        srcdir = "C:\Users\Micha\miniconda3\envs\test-meson"
        userbase = "C:\Users\Micha\AppData\Roaming\Python"
```

Reply:
Ok, I've worked this out. For some (stupid) reason, Meson never tries to use Python's `sysconfig` library to locate the development libraries or include dirs. Meson has 2 lookup modes for the Python dependency:
1. `pkgconfig`: invokes pkg-config to locate the libraries. Will fail if pkg-config is not installed (unlikely on Linux, but likely on macOS and Windows), if `PKG_CONFIG_PATH` is not properly set, or if the Python install does not have a `.pc` file
2. `system`: uses variables in `sysconfig` to build the expected library name and searches for that library in recognized compiler search paths (e.g., `LD_LIBRARY_PATH` on UNIX). This is why I said Meson's a bit stupid because it's already using `sysconfig` to build library names, but it's not using the `sysconfig` variables that directly point to the recognized libpython

So, because we build extension modules (i.e., the Cython code), users need one of the following to be true:
1. `pkg-config` must be installed
2. Library search variables (e.g., `LD_LIBRARY_PATH`) need to be properly set

Reply:
@michaelmckinsey1 when you get the chance, can you try at least one of these solutions on Windows? I'll try on macOS. If this works, I'll try to see if there's a way I can add pkg-config to the build dependencies